webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -428,22 +451,166 @@ export class ContactsComponent implements OnInit, OnDestroy{
     return contact._id + contact._rev;
   }
 
-  // const setActionBarData = () => {
-  //   ctrl.setLeftActionBar({
-  //     hasResults: ctrl.hasContacts,
-  //     userFacilityId: usersHomePlace && usersHomePlace._id,
-  //     childPlaces: allowedChildPlaces,
-  //     exportFn: function() {
-  //       Export('contacts', ctrl.filters, { humanReadable: true });
-  //     },
-  //   });
-  // };
-
-  // const updateAllowedChildPlaces = () => {
-  //   XmlForms.listen('ContactsListCtrl',{ contactForms: true }, (err, forms) => {
-  //     const allowCreateLink = contactType => forms && forms.find(form => form._id === contactType.create_form);
-  //     allowedChildPlaces = childPlaces.filter(allowCreateLink);
-  //     setActionBarData();
-  //   });
-  // };
+  private setLeftActionBar() {
+    this.globalActions.setLeftActionBar({
+      hasResults: this.hasContacts,
+      userFacilityId: this.usersHomePlace && this.usersHomePlace._id,
+      childPlaces: this.allowedChildPlaces?.sort((a, b) => a.id.localeCompare(b.id)),
+      exportFn: () => {
+        this.exportService.export('contacts', this.filters, { humanReadable: true });
+      }
+    });
+  }
+
+  private subscribeToSelectedContact() {
+    const subscription = this.store
+      .select(Selectors.getSelectedContact)
+      .subscribe((selected) => {
+        this.selectedContact = selected;
+        this.setChildTypesBySelectedContact();
+
+        this.globalActions.setRightActionBar({
+          relevantForms: [], // This disables the "New Action" button in action bar until forms load
+          sendTo: this.selectedContact?.type?.person ? this.selectedContact?.doc : '',
+          canDelete: !!this.selectedContact?.children?.every(group => !group.contacts?.length),
+          canEdit: this.sessionService.isAdmin() || this.userSettings?.facility_id === this.selectedContact?.doc?._id,
+          openContactMutedModal: (event, form) => this.openContactMutedModal(form),
+          openSendMessageModal: (sendTo) => this.openSendMessageModal(sendTo)
+        });
+
+        this.subscribeToAllContactXmlForms();
```

Comment:
Does this mean that the RHS actionbar contact forms are updated every time a new contact is selected? If yes, we should find a way to avoid this pointless update.

Reply:
yeah, it was like that in the original version. I think because of the filtering by the selected contact's child types

Reply:
Are you sure? 

Looking at the code, this function `updateAllowedChildPlaces()` is only called once, when the contacts-controller is created: https://github.com/medic/cht-core/blob/f20cf655da6bf38eea98bb25944619e03cf181ba/webapp/src/js/controllers/contacts.js#L310. I don't think it got called every time the selected contact changed.

Reply:
As it stands, I think the LHS actionbar shouldn't be updated when we select a new contact. And I just noticed that I said "RHS" in the first comment when I meant LHS 🤦🏻 :

Reply:
When setting a selected contact: https://github.com/medic/cht-core/blob/f20cf655da6bf38eea98bb25944619e03cf181ba/webapp/src/js/actions/contacts.js#L230